src/features/user/TreeMapper/components/PlantLocationPage.tsx

```diff
@@ -7,10 +7,15 @@ import { getFormattedNumber } from '../../../../utils/getFormattedNumber';
 import dynamic from 'next/dynamic';
```

Comment:
Here `DetailsProps` is defined as:
`interface DetailsProps {
  setselectedLocation: Function;
  location: Object;
}`
Instead of adding generic type declaration could you define it as:
` location: PlantLocation | SamplePlantLocation | null;
    setselectedLocation: SetState<PlantLocation | SamplePlantLocation | null>
  `
 After doing this, some new errors are generated in the file, please look into those.